beacon-chain/rpc/prysm/v1alpha1/validator/proposer_deneb.go

```diff
@@ -1,12 +1,59 @@
 package validator
 
 import (
+	"sync"
+
 	"github.com/prysmaticlabs/prysm/v4/consensus-types/interfaces"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
 	enginev1 "github.com/prysmaticlabs/prysm/v4/proto/engine/v1"
 	ethpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
 	"github.com/prysmaticlabs/prysm/v4/runtime/version"
 )
 
+var bundleCache = &blobsBundleCache{
+	blobs: make(map[primitives.Slot]*enginev1.BlobsBundle),
+}
+
+// BlobsBundleCache holds the KZG commitments and other relevant sidecar data for a local beacon block.
+type blobsBundleCache struct {
+	blobs map[primitives.Slot]*enginev1.BlobsBundle
+	sync.Mutex
+}
+
+// add adds a blobs bundle to the cache.
+// same slot overwrites the previous bundle.
+func (c *blobsBundleCache) add(slot primitives.Slot, bundle *enginev1.BlobsBundle) {
+	c.Lock()
+	c.blobs[slot] = bundle
+	c.Unlock()
+
+	// Trigger pruning in the background
+	go c.prune(slot)
```

Comment:
If you prune everything less than `slot` as soon as you get or set a value for slot, don't you "instantly" remove the previous slot, making this cache size of one? If you want cache size 1, it would seem simpler to me to define `blobBundleCache` as:
```
type blobsBundleCache struct {
	sync.Mutex
        slot primitives.Slot
	bundle*enginev1.BlobsBundle
}

func (c *blobsBundleCache) add(slot primitives.Slot, bundle *enginev1.BlobsBundle) {
	c.Lock()
	defer c.Unlock()
	c.bundle = bundle
	c.slot = slot
}

func (c *blobsBundleCache) get(slot primitives.Slot) *enginev1.BlobsBundle {
	c.Lock()
	defer c.Unlock()
	if c.slot == slot {
	    return c.bundle
	}
	return nil
}
```